include/libtorrent/torrent_handle.hpp

```diff
@@ -873,9 +871,12 @@ namespace libtorrent
 			, std::string const& private_key
 			, std::string const& dh_params);
 
+#ifndef TORRENT_NO_DEPRECATE
 		// Returns the storage implementation for this torrent. This depends on the
 		// storage constructor function that was passed to add_torrent.
+		TORRENT_DEPRECATED
 		storage_interface* get_storage_impl() const;
```

Comment:
I'm familiar with the problematic since providing a binding from java force me to do the two-phase initialization (not good).

I will undeprecate this and get the rest of the PR accepted, let's see if in the future another solution comes up.